### PR TITLE
[gtest] Upgrade to 1.11.0. Contributes to JB#54442

### PR DESCRIPTION
**Pull request by pbmedved**

### CODE REVIEW

**Review by okodron — COMMENTED** [review]
Better to use `%autosetup` instead of `%setup`

File: rpm/gtest.spec [hunk 1]
```diff
@@ -76,12 +79,14 @@ Requires:       libgmock = %{version}-%{release}
 Documentation files for libgmock.
 
 %prep
-%setup -q -n %{name}-%{version}/%{name}
+%autosetup -q -n %{name}-%{version}/%{name}
```
Comment (okodron) [inline rpm/gtest.spec]: In such case you need to pass additional argument for patch:
```suggestion
%autosetup -p1 -n %{name}-%{version}/%{name}
```
And you don't need any more to specify all the patches explicitly: https://rpm.org/user_doc/autosetup.html

**Review by okodron — CHANGES_REQUESTED** [review]

**Review by okodron — COMMENTED** [review]
As the main package is called `gtest` it sounds logical to rename `libgmock` sub-package to `gmock`. It will also comply with Fedora's version.

File: rpm/gtest.spec [hunk 2]
```diff
@@ -120,7 +119,7 @@ sed -e "s/set(GOOGLETEST_VERSION .*)/set(GOOGLETEST_VERSION %{version})/" -i CMa
 %doc googletest/samples
 
 %files -n libgmock
```
Comment (okodron) [inline rpm/gtest.spec]: Should be also renamed to `gmock`.

Comment (pbmedved) [reply rpm/gtest.spec]: done

**Review by okodron — APPROVED** [review]
LGTM

**Comment by pvuorela** [comment]
No objections on that gmock renaming per se, especially if Fedora isn't using the "lib" prefix, but looks like sailfish-browser is depending on libgmock-devel. Need to either fix that at the same time or then provide also the earlier name. Simplest might be to do the renaming separately.

Btw looks like there also 1.11.0 available now.

**Comment by okodron** [comment]
Can you change the MR title? Cause it's no more 1.10.0 upgrade...

File: rpm/gtest.spec [hunk 3]
```diff
@@ -119,19 +117,18 @@ sed -e "s/set(GOOGLETEST_VERSION .*)/set(GOOGLETEST_VERSION %{version})/" -i CMa
 %doc googletest/docs/
 %doc googletest/samples
 
-%files -n libgmock
-%license googlemock/LICENSE
+%files -n gmock
+%license ./LICENSE
```
Comment (okodron) [inline rpm/gtest.spec]: Just this:
```suggestion
%license LICENSE
```

File: rpm/gtest.spec [hunk 4]
```diff
@@ -98,11 +96,11 @@ sed -e "s/set(GOOGLETEST_VERSION .*)/set(GOOGLETEST_VERSION %{version})/" -i CMa
...
 %postun -p /sbin/ldconfig
 
-%post -n libgmock -p /sbin/ldconfig
-%postun -n libgmock -p /sbin/ldconfig
+%post -n gmock -p /sbin/ldconfig
+%postun -n gmock -p /sbin/ldconfig
 
 %files
-%license googletest/LICENSE
+%license ./LICENSE
```
Comment (okodron) [inline rpm/gtest.spec]:
```suggestion
%license LICENSE
```

**Review by okodron — CHANGES_REQUESTED** [review]

**Review by okodron — APPROVED** [review]

**Comment by pvuorela** [comment]
Looks nicely simple update now, but for making sure I tried doing a test build in OBS. However this happened:
```
[   26s] CMake Error at googlemock/CMakeLists.txt:43 (project):
[   26s]   VERSION "1.8.1+1.10.0.20210618151355.1.g0b64396" format invalid.
```

The .spec has sed call for replacing the version. Looks suspicious and would expect it's not needed.

**Comment by okodron** [comment]
> Looks nicely simple update now, but for making sure I tried doing a test build in OBS. However this happened:
> 
> ```
> [   26s] CMake Error at googlemock/CMakeLists.txt:43 (project):
> [   26s]   VERSION "1.8.1+1.10.0.20210618151355.1.g0b64396" format invalid.
> ```
> 
> The .spec has sed call for replacing the version. Looks suspicious and would expect it's not needed.

Can it happen due to the "not-so-expected" version from your local build? I mean whether it works with `1.11.1` version set via tag?

**Comment by martyone** [comment]
The .spec shouldn't use `%{version}` as is. It should do something like https://github.com/sailfishos/libqofono/blob/master/rpm/libqofono-qt5.spec#L60 to allow test builds from untagged revisions.

**Comment by pvuorela** [comment]
Yes, something like that might do the trick. Then again grepping from the sources there's already
`CMakeLists.txt:set(GOOGLETEST_VERSION 1.11.0)` and don't see why we should be overriding that in the build.

**Comment by martyone** [comment]
I see two issues:
1. unnecessary modifying sources during build
2. hardcoding (duplicating) version in sources

I would omit setting the value in CMakeLists.txt and instead assert there that it was provided on cmake command line (possibly falling back to some default like 1.0 for convenience).

**Comment by okodron** [comment]
> Yes, something like that might do the trick. Then again grepping from the sources there's already
> `CMakeLists.txt:set(GOOGLETEST_VERSION 1.11.0)` and don't see why we should be overriding that in the build.

Indeed: version patching seems like an unnecessary step at least now.

**Review by okodron — CHANGES_REQUESTED** [review]
@pbmedved, please, drop the version patching with sed.

**Comment by pbmedved** [comment]
It should build normally now.

**Comment by pvuorela** [comment]
Better but then hit another problem:
```
[   41s] RPM build errors:
[   41s]     File not found: /home/abuild/rpmbuild/BUILDROOT/gtest-1.8.1+1.10.0.20210622105856.1.ga3229a7-1.2.1.jolla.i386/usr/lib/libgtest.so.1.8.1+1.10.0.20210622105856.1.ga3229a7
[   41s]     File not found: /home/abuild/rpmbuild/BUILDROOT/gtest-1.8.1+1.10.0.20210622105856.1.ga3229a7-1.2.1.jolla.i386/usr/lib/libgtest_main.so.1.8.1+1.10.0.20210622105856.1.ga3229a7
```

The %files section in main an gmock has %{version} appended. That's just silly, replace with so.*

**Review by okodron — COMMENTED** [review]
Looks better now.

**Comment by pvuorela** [comment]
Now builds and libphonenumber on top of this. Browser on obs didn't like having a choice on gmock devel for the old and new names, but I'll do a local build to be on the safe side.

**Comment by pvuorela** [comment]
And local build fine. LGTM!

**Comment by pvuorela** [comment]
Didn't fully work. https://github.com/sailfishos/gtest/pull/4/files

**Pull request merged**